src/ImageSharp/ColorSpaces/Companding/SRgbCompanding.cs

```diff
@@ -48,27 +46,23 @@ public static void Compress(Span<Vector4> vectors)
             for (int i = 0; i < vectors.Length; i++)
             {
                 ref Vector4 v = ref Unsafe.Add(ref baseRef, i);
-                v.X = Compress(v.X);
-                v.Y = Compress(v.Y);
-                v.Z = Compress(v.Z);
+                Compress(ref v);
             }
         }
 
         /// <summary>
         /// Expands a companded vector to its linear equivalent with respect to the energy.
         /// </summary>
         /// <param name="vector">The vector.</param>
-        /// <returns>The <see cref="Vector4"/> representing the linear channel values.</returns>
         [MethodImpl(InliningOptions.ShortMethod)]
-        public static Vector4 Expand(Vector4 vector) => new Vector4(Expand(vector.X), Expand(vector.Y), Expand(vector.Z), vector.W);
+        public static void Expand(ref Vector4 vector) => vector = new Vector4(Expand(vector.X), Expand(vector.Y), Expand(vector.Z), vector.W);
```

Comment:
The unnecessary constructor invocation + copy is still there. What I meant is:

```C#
void Expand(ref Vector4 v) 
{
    v.X =Expand(v.X);
    v.Y =Expand(v.Y);
    v.Z =Expand(v.Z);
    v.W =Expand(v.W);    
}
```

It's unlikely that JIT is capable to optimize the simple version. Try a benchmark for a proof/disproof! :smile:

Reply:
Yeah, I got it. Already committed a change locally. 👍